File: TwoSumII-InputArrayIsSorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l = 0
        r = len(numbers) - 1

        while l < r:
            tar = numbers[l] + numbers[r]
            if tar == target:
                return [l+1, r+1]
            elif tar < target:
                l += 1
            else:
                r -= 1

    # Time: O(n)
    # Space: O(1)

File: TwoSumII-InputArrayIsSorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l = 0;
        int r = numbers.length - 1;
        int[] ans = new int[2];

        while(l < r){
            int curr = numbers[l] + numbers[r];
            if (curr == target){
                ans[0] = l + 1;
                ans[1] = r + 1;
                return ans;
            }
            else if(curr < target){
                l++;
            }
            else{
                r--;
            }
        }
        return ans;
    }
    //Time: O(n)
    //Space: O(1)
}
